src/components/dialogs/generator-creation-dialog.js

```diff
@@ -82,17 +85,36 @@ const GeneratorCreationDialog = ({
 
     const inputForm = useInputForm();
 
+    const [formValues, setFormValues] = useState(undefined);
+
+    const [isDialogSearchOpen, setDialogSearchOpen] = useState(false);
+
+    const handleCloseSearchDialog = () => {
+        setDialogSearchOpen(false);
+    };
+
+    const handleOpenSearchDialog = () => {
+        setDialogSearchOpen(true);
+    };
+
+    const copyEquipmentButton = useButtonWithTooltip({
+        label: 'CopyFromExisting',
+        handleClick: handleOpenSearchDialog,
+    });
+
```

Comment:
these lines are repeated in each forms, can't we use a custom hook ?

Reply:
fixed